app/src/main/java/org/schabi/newpipe/util/ListHelper.java

```diff
@@ -309,28 +304,47 @@ static int getHighestQualityAudioIndex(@Nullable MediaFormat format,
      * @param audioStreams List of audio streams
      * @return Index of audio stream that produces the most compact results or -1 if not found
      */
-    static int getMostCompactAudioIndex(@Nullable MediaFormat format,
-                                        final List<AudioStream> audioStreams) {
-        int result = -1;
-        if (audioStreams != null) {
-            while (result == -1) {
-                AudioStream prevStream = null;
-                for (int idx = 0; idx < audioStreams.size(); idx++) {
-                    final AudioStream stream = audioStreams.get(idx);
-                    if ((format == null || stream.getFormat() == format)
-                            && (prevStream == null || compareAudioStreamBitrate(prevStream, stream,
-                                    AUDIO_FORMAT_EFFICIENCY_RANKING) > 0)) {
-                        prevStream = stream;
-                        result = idx;
-                    }
-                }
-                if (result == -1 && format == null) {
-                    break;
-                }
-                format = null;
+    static int getMostCompactAudioIndex(@Nullable final MediaFormat format,
+                                        @Nullable final List<AudioStream> audioStreams) {
+
+        return getAudioIndexByHighestRank(format, audioStreams,
+                // The "-" is important -> Compares ascending (first = highest rank)
+                (s1, s2) -> -compareAudioStreamBitrate(s1, s2, AUDIO_FORMAT_EFFICIENCY_RANKING)
+        );
+    }
+
+    /**
+     * Get the audio-stream from the list with the highest rank, depending on the comparator.
+     * Format will be ignored if it yields no results.
+     *
+     * @param targetedFormat The target format type or null if it doesn't matter
+     * @param audioStreams   List of audio streams
+     * @param comparator     The comparator used for determining the max/best/highest ranked value
+     * @return Index of audio stream that produces the most compact results or -1 if not found
```

Comment:
I think you forgot to update the `@return` description when copy-pasting the javadoc ;-)

Reply:
@litetex ^

Reply:
Done